ci/appveyor-cpp-build.bat

```diff
@@ -95,11 +95,12 @@ if "%JOB%" == "Build_Debug" (
   exit /B 0
 )
 
-conda create -n arrow -q -y ^
+conda create -n arrow -q -y -c conda-forge ^
+      --file=ci\conda_env_python.yml ^
       python=%PYTHON% ^
-      six pytest setuptools numpy pandas cython hypothesis ^
-      thrift-cpp=0.11.0 boost-cpp ^
-      -c conda-forge
+      numpy=1.14.0 ^
```

Comment:
Maybe pin here to `=1.14` to get bugfix releases and new builds.

Reply:
One line below thrift was pinned to patch version too. Pin to minor?

Reply:
Yes. That will be especially helpful as `conda-forge` is also currently doing the compiler migration just as we write.